src/codegen/mod.rs

```diff
@@ -1302,14 +1302,15 @@ impl<'a> FieldCodegen<'a> for Bitfield {
                     let val = val as $bitfield_int_ty as $unit_field_int_ty;
 
                     let mut unit_field_val: $unit_field_int_ty = unsafe {
-                        ::$prefix::mem::transmute(self.$unit_field_ident)
+                        *(&self.$unit_field_ident as *const _ as *const $unit_field_int_ty)
                     };
                     unit_field_val &= !mask;
                     unit_field_val |= (val << $offset) & mask;
 
-                    self.$unit_field_ident = unsafe {
-                        ::$prefix::mem::transmute(unit_field_val)
-                    };
+                    unsafe {
+                        self.$unit_field_ident = ::$prefix::mem::zeroed();
+                        *(&mut self.$unit_field_ident as *mut _ as *mut $unit_field_int_ty) |= unit_field_val;
```

Comment:
I checked with `#rust-internals`, and even though this compiles right now to sane code, it can lead to UB because of unaligned stores, so I'll need to find a smarter way of writing this before landing I guess :(